tools/splat_ext/pm_sbn.py

```diff
@@ -0,0 +1,738 @@
+            id = mseq.get("id", "auto")
+            if id == "auto":
+                id = len(self.init.mseq_entries)
+            assert type(id) == int
+
+            # Lookup file ID
+            file = mseq.get("file")
+            assert type(file) == str
+            file_id = self.lookup_file_id(file)
+
+            # Replace self.init.mseq_entries[id]
+            if id < len(self.init.mseq_entries):
+                print(f"Overwriting MSEQ ID {id:02X}")
+                self.init.mseq_entries[id] = file_id
+            elif id == len(self.init.mseq_entries):
+                self.init.mseq_entries.append(file_id)
+            else:
+                raise Exception(f"Invalid MSEQ ID: 0x{id:02X} - cannot have gaps")
+
+        # Read INIT banks
+        for bank in banks:
+            id = bank.get("id", "auto")
+            if id == "auto":
+                id = len(self.init.bk_entries)
+            assert type(id) == int
+
+            file = bank.get("file")
+            assert type(file) == str
+            file_id = self.lookup_file_id(file)
+
+            bank_index = bank.get("bank_index")
+            assert type(bank_index) == int
+
+            bank_group = bank.get("bank_group")
+            assert type(bank_group) == int
+
+            entry = BufferEntry()
+            entry.fileIndex = file_id
+            entry.bankIndex = bank_index
+            entry.bankGroup = bank_group
+
+            # Replace self.init.bk_entries[id]
+            if id < len(self.init.bk_entries):
+                print(f"Overwriting bank ID {id:02X}")
+                self.init.bk_entries[id] = entry
+            elif id == len(self.init.bk_entries):
+                self.init.bk_entries.append(entry)
+            else:
+                raise Exception(f"Invalid bank ID: {id:02X} - cannot have gaps")
+
+        return [config_file_path]
+
+
+    def lookup_file_id(self, filename: str) -> int:
+        for id, sbn_file in enumerate(self.files):
+            if sbn_file.file_name() == filename:
+                return id
+        raise Exception(f"File not found: {filename} - is it in the file_id_map?")
+
+    def __str__(self) -> str:
+        s = "SBN(\n"
+        s += "  files=[\n"
+        for sbn_file in self.files:
+            s += f"    {sbn_file},\n"
+        s += "  ],\n"
+        s += f"  init={self.init}\n"
+        s += ")"
+        return s
+
+    def __hash__(self) -> int:
+        h = 0
+        for sbn_file in self.files:
+            h ^= hash(sbn_file)
+        h ^= hash(self.init)
+        return h
+
+class SBNHeader:
+    signature: bytes
+    size: int
+    unk_08: int
+    unk_0C: int
+    tableOffset: int
+    numEntries: int
+    unk_18: int
+    unk_1C: int
+    unk_20: int
+    INIToffset: int
+    unk_28: int
+    unk_2C: int
+    unk_30: int
+    unk_34: int
+    unk_38: int
+    unk_3C: int
+
+    def decode(self, data: bytes) -> int:
+        self.signature, = struct.unpack(">4s", data[0x00:0x04])
```

Comment:
If we make this a dataclass, you can make a from_bytes that looks like

```
@staticmethod
def from_bytes(data: bytes) -> SBNHeader:
    return SBNHeader(*struct.unpack(">4siiiiiiiiiiiiiiii"))
```
to avoid manually setting each specific field and calling unpack a bunch of times in a row